### PR TITLE
Replace "data" with "dAtA" in generated code.

### PR DESCRIPTION
**Pull request by johanbrandhorst**
This allows the use of the proto package name "data" without conflicts in generated code.

Fixes issue #208. I tried building the tests but was getting unrelated errors, are there any other file changes needed?

### CODE REVIEW

**Comment by johanbrandhorst** [comment]
I am actually having trouble regenerating the files, so I've gone ahead and manually replaced all instances of data with dAtA in the generated files. Travis should find the files I've messed up if there's any left I hope.

**Comment by johanbrandhorst** [comment]
Alright, tests are passing, but considering how many files I had to change, I have a half a feeling I just changed the tests to be wrong. Please take a careful look at all the changes I made and preferrably rebuild everything in your environment as well.

File: protoc-gen-gogo/generator/generator.go [hunk 1]
```diff
@@ -547,7 +547,7 @@ func (es enumSymbol) GenerateAlias(g *Generator, pkg string) {
 	g.P("func (x ", s, ") String() string { return (", pkg, ".", s, ")(x).String() }")
 	if !es.proto3 {
 		g.P("func (x ", s, ") Enum() *", s, "{ return (*", s, ")((", pkg, ".", s, ")(x).Enum()) }")
-		g.P("func (x *", s, ") UnmarshalJSON(data []byte) error { return (*", pkg, ".", s, ")(x).UnmarshalJSON(data) }")
+		g.P("func (x *", s, ") UnmarshalJSON(dAtA []byte) error { return (*", pkg, ".", s, ")(x).UnmarshalJSON(dAtA) }")
```
Comment (awalterschulze) [inline protoc-gen-gogo/generator/generator.go]: Is this change really necessary?

File: protoc-gen-gogo/generator/generator.go [hunk 2]
```diff
@@ -1573,9 +1573,9 @@ func (g *Generator) generateEnum(enum *EnumDescriptor) {
 		g.P("}")
 	}
 	if !enum.proto3() {
-		g.P("func (x *", ccTypeName, ") UnmarshalJSON(data []byte) error {")
+		g.P("func (x *", ccTypeName, ") UnmarshalJSON(dAtA []byte) error {")
```
Comment (awalterschulze) [inline protoc-gen-gogo/generator/generator.go]: And this one?

File: protoc-gen-gogo/generator/generator.go [hunk 3]
```diff
@@ -1573,9 +1573,9 @@ func (g *Generator) generateEnum(enum *EnumDescriptor) {
...
 		g.In()
-		g.P("value, err := ", g.Pkg["proto"], ".UnmarshalJSONEnum(", ccTypeName, `_value, data, "`, ccTypeName, `")`)
+		g.P("value, err := ", g.Pkg["proto"], ".UnmarshalJSONEnum(", ccTypeName, `_value, dAtA, "`, ccTypeName, `")`)
```
Comment (awalterschulze) [inline protoc-gen-gogo/generator/generator.go]: And this one?

**Comment by awalterschulze** [comment]
The rest looks good :)

File: protoc-gen-gogo/generator/generator.go (hunk 3 above)
Comment (awalterschulze) [inline protoc-gen-gogo/generator/generator.go]: Ok so you are saying this one is necessary? I just want to be certain.

Comment (johanbrandhorst) [inline protoc-gen-gogo/generator/generator.go]: I thought I deleted that one as well, I really don't know what I'm doing here.

**Comment by johanbrandhorst** [comment]
I expect I might have to update some tests but I'm waiting for Travis to fail to see which ones

File: protoc-gen-gogo/generator/generator.go (hunk 3 above)
Comment (awalterschulze) [inline protoc-gen-gogo/generator/generator.go]: my bad

**Comment by awalterschulze** [comment]
I think the tests will pass :)

**Comment by awalterschulze** [comment]
I think the last thing this needs is a test in the test folder.
So say ./test/data with a proto file that has a package name data and a single message.
You should then also add a Makefile to regenerate this proto
And add a command for this in the main Makefile
Please see the other examples in the test folder.

**Comment by johanbrandhorst** [comment]
I've just tried that, substituted with my fork of course:

```
go install github.com/johanbrandhorst/protobuf/protoc-min-version
go install github.com/johanbrandhorst/protobuf/protoc-gen-gogo
protoc-min-version --version="3.0.0" --proto_path=../../../../../:../../protobuf/:. --gogo_out=. data.proto
```

It creates protofiles, but not with `dAtA` anywhere, so it just doesn't seem to be using the correct plugin. No idea why. Any chance you could do the last few things here? Sorry to be a bother.

**Comment by johanbrandhorst** [comment]
Ok I've checked in the files that I generated, will wait for Travis to fail so I can see what I need to fix still.

**Comment by awalterschulze** [comment]
maybe just run `which protoc-gen-gogo`
I think there might be a stale one in your PATH

**Comment by johanbrandhorst** [comment]
I've tried deleting the old one, confirming that it was removed, rebuilding it, confirming that it exists again, and it still generates old code. I have no idea why, I spent some time looking at it during the weekend I submitted this and couldn't figure it out so I've just resorted to using Travis to find my problems. I know it's stupid, but at least that way I can interleave it with work while I'm at work.

**Comment by johanbrandhorst** [comment]
Ha, it passed 🏄

File: test/data/Makefile [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# Protocol Buffers for Go with Gadgets
+#
+# Copyright (c) 2013, The GoGo Authors. All rights reserved.
```
Comment (awalterschulze) [inline test/data/Makefile]: 2016

File: test/data/data.proto [hunk 5]
```diff
@@ -0,0 +1,49 @@
+// Protocol Buffers for Go with Gadgets
+//
+// Copyright (c) 2013, The GoGo Authors. All rights reserved.
```
Comment (awalterschulze) [inline test/data/data.proto]: 2016

File: test/data/data_test.go [hunk 6]
```diff
@@ -0,0 +1,40 @@
+// Protocol Buffers for Go with Gadgets
+//
+// Copyright (c) 2013, The GoGo Authors. All rights reserved.
```
Comment (awalterschulze) [inline test/data/data_test.go]: Is this file really necessary?

**Comment by awalterschulze** [comment]
You can add yourself and/or your company to the authors / contributors as appropriate.

File: test/data/data_test.go (hunk 6 above)
Comment (johanbrandhorst) [inline test/data/data_test.go]: Not really, I just looked at the other files and they all seemed to have one of these tests, I'll remove it.

**Comment by awalterschulze** [comment]
Thank you for you efforts :)

**Comment by johanbrandhorst** [comment]
Woho! Thanks for the amazing project!

**Comment by awalterschulze** [comment]
Thanks for your support and contributing :)
